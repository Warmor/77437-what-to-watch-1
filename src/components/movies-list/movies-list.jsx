import React from 'react';
import PropTypes from 'prop-types';

import SmallMovieCard from '~/components/small-movie-card/small-movie-card';
import withActiveFilm from "~/hocs/with-active-film/with-active-film";
const SmallMovieCardWrap = withActiveFilm(SmallMovieCard);

function MoviesList(props) {
  const {
    films,
  } = props;
  return (
    <div className="catalog__movies-list">
      {films.map((film) => {
        return <SmallMovieCardWrap
          key={film.id}
          film={film}
        />;
      })}
    </div>
  );
}

MoviesList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default MoviesList;
