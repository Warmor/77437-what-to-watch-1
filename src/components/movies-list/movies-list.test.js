import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
Enzyme.configure({adapter: new Adapter()});

import {MoviesList} from "~/components/movies-list/movies-list";

describe(`<MoviesList/>`, () => {
  it(`MoviesList renders correctly`, () => {
    const mockProps = {
      films: [{
        id: 1,
        title: `Fantastic Beasts`,
        genreId: 1,
        image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      }],
    };
    const tree = shallow(<MoviesList films={mockProps.films}/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

