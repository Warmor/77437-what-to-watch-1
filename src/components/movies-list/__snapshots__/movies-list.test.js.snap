// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesList/> MoviesList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <WithActiveFilm
    film={
      Object {
        "genreId": 1,
        "id": 1,
        "image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "title": "Fantastic Beasts",
        "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      }
    }
    key="1"
  />
</div>
`;
